cdap-common/src/main/java/co/cask/cdap/common/utils/Networks.java

```diff
@@ -94,4 +96,15 @@ public static String normalizeWebappDiscoveryName(String name) throws Unsupporte
 
     return URLEncoder.encode(name, Charsets.UTF_8.name());
   }
+
+
+  /**
+   * Get the filename part from a {@link URI}. For a URI "file:///tmp/foo.jar", "foo.jar" will be returned.
+   *
+   * @param uri The {@link URI} to get filename from.
+   * @return filename.
+   */
+  public static String getURIFilename(URI uri) {
+    return new File(uri.getPath()).getName();
```

Comment:
To remove this method or just not to call it in `LocalizationUtils.getLocalizedName()`?